bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusRelayWithInputHandler.java

```
@@ -0,0 +1,131 @@
+
+/**
+ * The {@link VelbusRelayWithInputHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniel Rosengarten - Initial contribution
+ */
+@NonNullByDefault
+public class VelbusRelayWithInputHandler extends VelbusRelayHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(Arrays.asList(THING_TYPE_VMB1RYS));
+
+    private static final StringType PRESSED = new StringType("PRESSED");
+    private static final StringType LONG_PRESSED = new StringType("LONG_PRESSED");
+
+    public VelbusRelayWithInputHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+
+        VelbusBridgeHandler velbusBridgeHandler = getVelbusBridgeHandler();
+        if (velbusBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
+        }
+
+        if (isButtonChannel(channelUID) && command instanceof StringType) {
+            StringType stringTypeCommand = (StringType) command;
+
+            if (stringTypeCommand.equals(PRESSED) || stringTypeCommand.equals(LONG_PRESSED)) {
+                VelbusButtonPacket packet = new VelbusButtonPacket(getModuleAddress().getChannelIdentifier(channelUID));
+
+                try {
+                    packet.Pressed();
+                    velbusBridgeHandler.sendPacket(packet.getBytes());
+                    Thread.sleep(20);
+
+                    if (stringTypeCommand.equals(LONG_PRESSED)) {
+                        packet.LongPressed();
+                        velbusBridgeHandler.sendPacket(packet.getBytes());
+                        Thread.sleep(20);
+                    }
+
+                    packet.Released();
+                    velbusBridgeHandler.sendPacket(packet.getBytes());
+                } catch (InterruptedException e) {
```

This empty catch clause will probably become obsolete if you use the scheduler instead of a Thread.sleep.

---

I d'idn't check this part of your code, the 20 ms delay is useless then. I've remove this from my code.